mmv1/third_party/terraform/services/alloydb/resource_alloydb_cluster_test.go

```diff
@@ -1295,3 +1295,39 @@ resource "google_compute_network" "default" {
 data "google_project" "project" {}
 `, context)
 }
+
+// Ensures cluster creation succeeds for a Private Service Connect enabled cluster.
+func TestAccAlloydbCluster_withPrivateServiceConnect(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckAlloydbClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
```

Comment:
could you add a step that disables psc as well?

Reply:
PSC cannot be disabled, once it is enabled. That is, a cluster that has PSC enabled at the time of creation can never disable it in future.